### PR TITLE
Kernel/VirtIO: Ensure proper error propagation in core methods

### PR DESCRIPTION
**Pull request by supercomputer7**
Taken as a task from this comment ([here](https://github.com/SerenityOS/serenity/pull/19757#discussion_r1259259110), by @ADKaster).

### CODE REVIEW

**Review by Hendiadyoin1 — COMMENTED** [review]
If I do remember correctly casting from an errno to Error does not need the `from_errno` helper in the kernel

**Comment by supercomputer7** [comment]
> If I do remember correctly casting from an errno to Error does not need the `from_errno` helper in the kernel

It's correct, but it's more clear this way (and we do that in many places in the Kernel already).

File: Kernel/Bus/VirtIO/Device.cpp [hunk 1]
```diff
@@ -114,14 +110,14 @@ bool Device::setup_queues(u16 requested_queue_count)
 
     dbgln_if(VIRTIO_DEBUG, "{}: Setting up {} queues", m_class_name, m_queue_count);
     for (u16 i = 0; i < m_queue_count; i++) {
-        if (!setup_queue(i))
-            return false;
+        if (auto error_or_void = setup_queue(i); error_or_void.is_error())
+            return error_or_void.release_error();
```
Comment (ADKaster) [inline Kernel/Bus/VirtIO/Device.cpp:114]: this and below are just `TRY(setup_queue(i)` right?

**Review by ADKaster — COMMENTED** [review]
Looks nice, one nitpick

File: Kernel/Bus/VirtIO/Device.cpp [hunk 2]
```diff
@@ -113,15 +109,13 @@ bool Device::setup_queues(u16 requested_queue_count)
...
-        if (!m_transport_entity->activate_queue({}, i))
-            return false;
-    }
-    return true;
+    for (u16 i = 0; i < m_queue_count; i++)
+        TRY(setup_queue(i));
+
+    // NOTE: Queues can only be activated *after* all others queues were also configured
+    for (u16 i = 0; i < m_queue_count; i++)
+        TRY(m_transport_entity->activate_queue({}, i));
```
Comment (ADKaster) [inline Kernel/Bus/VirtIO/Device.cpp:117]: Not going to block this PR but I am curious: What happens if we activate 14 out of 16 queues, for example, and then return with an error? 

Is the system in an inconsistent state? I think this kind of "lack of strong exception guarantee" (for lack of a better word) is a general problem in our TRY-heavy code.

**Review by ADKaster — APPROVED** [review]

File: Kernel/Bus/VirtIO/Device.cpp (hunk 2 above)
Comment (supercomputer7) [reply Kernel/Bus/VirtIO/Device.cpp:117]: For VirtIO devices, failing to initialize the device results in a flow where host software should indicate this explicitly by assigning failure status to the status bits, iirc.

**Pull request merged**